substrate-node/pallets/pallet-smart-contract/src/lib.rs

```diff
@@ -1603,8 +1601,10 @@ impl<T: Config> Pallet<T> {
             return;
         }
 
-        // Save the contract to be billed in (now -1 %(mod) BILLING_FREQUENCY_IN_BLOCKS)
-        let index = Self::get_contract_index().checked_sub(1).unwrap_or(0);
```

Comment:
With this calculation we never get to the last index of `ContractsToBillAt` which is `billing_frequency - 1`